File: web/src/api/follow-me.ts
import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query"
import { apiClient } from "@/lib/api-client"
import { useAuthStore } from "@/stores/auth-store"
import { queryKeys } from "./query-keys"

// Types matching backend FollowMeResponse
export interface FollowMeDestination {
  id: string
  position: number
  destination: string
  ring_time: number
}

export interface FollowMeDestinationData {
  destination: string
  ring_time?: number
}

export interface FollowMe {
  id: string
  tenant_id: string
  extension_id: string
  enabled: boolean
  strategy: string
  ring_extension_first: boolean
  extension_ring_time: number
  destinations: FollowMeDestination[]
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface FollowMeUpdate {
  enabled?: boolean
  strategy?: string
  ring_extension_first?: boolean
  extension_ring_time?: number
  destinations?: FollowMeDestinationData[]
}

export function useFollowMe(extensionId: string) {
  const tenantId = useAuthStore((s) => s.activeTenantId)!
  return useQuery({
    queryKey: queryKeys.followMe.detail(tenantId, extensionId),
    queryFn: () => apiClient.get<FollowMe>(`tenants/${tenantId}/extensions/${extensionId}/follow-me`),
    enabled: !!tenantId && !!extensionId,
  })
}

export function useUpdateFollowMe() {
  const tenantId = useAuthStore((s) => s.activeTenantId)!
  const qc = useQueryClient()
  return useMutation({
    mutationFn: ({ extensionId, ...data }: { extensionId: string } & FollowMeUpdate) =>
      apiClient.put<FollowMe>(`tenants/${tenantId}/extensions/${extensionId}/follow-me`, data),
    onSuccess: () => qc.invalidateQueries({ queryKey: queryKeys.followMe.all(tenantId) }),
  })
}

File: api/src/new_phone/schemas/follow_me.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from new_phone.models.follow_me import FollowMeStrategy


class FollowMeDestinationData(BaseModel):
    destination: str = Field(..., min_length=1, max_length=40)
    ring_time: int = Field(20, ge=5, le=120)


class FollowMeUpdate(BaseModel):
    enabled: bool = False
    strategy: FollowMeStrategy = FollowMeStrategy.SEQUENTIAL
    ring_extension_first: bool = True
    extension_ring_time: int = Field(25, ge=5, le=120)
    destinations: list[FollowMeDestinationData] = Field(default_factory=list, max_length=10)


class FollowMeDestinationResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    position: int
    destination: str
    ring_time: int


class FollowMeResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    tenant_id: uuid.UUID
    extension_id: uuid.UUID
    enabled: bool
    strategy: str
    ring_extension_first: bool
    extension_ring_time: int
    destinations: list[FollowMeDestinationResponse] = []
    is_active: bool
    created_at: datetime
    updated_at: datetime
